toolkit/tools/scheduler/scheduler.go

```diff
@@ -233,11 +202,14 @@ func cancelBuildsOnSignal(signals chan os.Signal, agent buildagents.BuildAgent)
 
 // buildGraph builds all packages in the dependency graph requested.
 // It will save the resulting graph to outputFile.
-func buildGraph(inputFile, outputFile string, agent buildagents.BuildAgent, workers, buildAttempts int, checkAttempts int, stopOnFailure, canUseCache bool, packagesToBuild, packagesToRebuild, ignoredPackages []*pkgjson.PackageVer, toolchainPackages []string, deltaBuild bool, allowToolchainRebuilds bool) (err error) {
+func buildGraph(inputFile, outputFile string, agent buildagents.BuildAgent, workers, buildAttempts int, checkAttempts int, stopOnFailure, canUseCache bool, packagesToBuild, packagesToRebuild, ignoredPackages []*pkgjson.PackageVer, toolchainPackages []string, optimizeWithCachedImplicit bool, allowToolchainRebuilds bool) (err error) {
 	// graphMutex guards pkgGraph from concurrent reads and writes during build.
 	var graphMutex sync.RWMutex
 
-	isGraphOptimized, pkgGraph, goalNode, err := schedulerutils.InitializeGraph(inputFile, packagesToBuild, deltaBuild)
+	// If optimizeWithCachedImplicit is true, we can use the cached implicit dependencies to aggressively prune the graph during the first pass. We will still
+	// try to avoid using the cached implicit dependencies until we have no other choice during the build, but since the graph is pruned, we will
+	// avoid building packages that are not needed.
+	_, pkgGraph, goalNode, err := schedulerutils.InitializeGraph(inputFile, nil, packagesToBuild, (canUseCache && optimizeWithCachedImplicit))
```

Comment:
>(canUseCache && optimizeWithCachedImplicit) [](http://example.com/codeflow?start=95&length=43)

Could we turn this `&&` to a variable? This way the variable's name will give us a helpful hint about the meaning of the arugment.